ui/de-lib/src/main/java/org/iplantc/de/commons/client/widgets/DETree.java

```diff
@@ -0,0 +1,67 @@
+package org.iplantc.de.commons.client.widgets;
+
+import com.google.common.base.Preconditions;
+
+import com.sencha.gxt.core.client.ValueProvider;
+import com.sencha.gxt.data.shared.TreeStore;
+import com.sencha.gxt.widget.core.client.tree.Tree;
+
+/**
+ * @author aramsey
+ *
+ * This is basically a clone-and-one of Sencha's Tree, but with the ability
+ * to define a way to set the debug IDs on the TreeNodes.
+ *
+ * A class will need to extend DETree and fill out the generateDebugId method
+ * The view that uses the class then has to call tree.ensureDebugId(id) in the
+ * view's onEnsureDebugId(String baseID) method like you would for any other
+ * UI component
+ */
+public abstract class DETree<M, C> extends Tree<M, C> {
+
+    private String baseId;
+
+    public DETree(TreeStore<M> store, ValueProvider<? super M, C> valueProvider) {
+        super(store, valueProvider);
+    }
+
+    public DETree(TreeStore<M> store,
+                  ValueProvider<? super M, C> valueProvider,
+                  TreeAppearance appearance) {
+        super(store, valueProvider, appearance);
+    }
+
+    public static class DETreeNode<M> extends TreeNode<M> {
+
+        protected DETreeNode(String modelId, M m, String domId) {
+            super(modelId, m, domId);
+        }
+    }
+
+    @Override
+    protected String register(M m) {
+        String id = generateModelId(m);
+        if (nodes.containsKey(id)) {
+            return nodes.get(id).getDomId();
+        } else {
+            String domId = generateDebugId(m);
```

Comment:
In Sencha's `Tree` class, the `register` method directly calls `XDOM.getUniqueId()` to create the `domId` for each TreeNode. So for `DETree` I removed that and defined a `generateDebugId` method that child classes can implement and set the IDs however they want.